File: Tests/SearchTests.fs
namespace Core

open System
open Stringier
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type SearchTests() =
    [<TestMethod>]
    member _.``Brute Force`` () =
        Assert.AreEqual(3, Search.BruteForce("helloworld", "low"))
        Assert.AreEqual(-1, Search.BruteForce("helloworld", "bacon"))

    [<TestMethod>]
    member _.``Rabin-Karp`` () =
        Assert.AreEqual(3, rabinKarp "helloworld" "low")
        Assert.AreEqual(-1, rabinKarp "helloworld" "bacon")

    [<TestMethod>]
    member _.``Boyer-Moore-Horspool`` () =
        Assert.AreEqual(3, horspool "helloworld" "low")
        Assert.AreEqual(-1, horspool "helloworld" "bacon")

File: Tests/SearchTests.cs
using System;
using Stringier;
using Xunit;

namespace Tests {
	public class SearchTests {
		[Theory]
		[InlineData("helloworld", "low", 3)]
		[InlineData("helloworld", "bacon", -1)]
		public void BruteForce(String source, String pattern, Int32 expected) => Assert.Equal(expected, Search.BruteForce(source, pattern));

		[Theory]
		[InlineData("helloworld", "low", 3)]
		[InlineData("helloworld", "bacon", -1)]
		public void RabinKarp(String source, String pattern, Int32 expected) => Assert.Equal(expected, Search.RabinKarp(source, pattern));

		[Theory]
		[InlineData("helloworld", "low", 3)]
		[InlineData("helloworld", "bacon", -1)]
		public void BoyerMooreHorspool(String source, String pattern, Int32 expected) => Assert.Equal(expected, Search.Horspool(source, pattern));
	}
}
